wetterdienst/core/point_data.py

```diff
@@ -4,25 +4,106 @@
 from enum import Enum
 from typing import Tuple, Union, Optional, List, Generator, Dict
 from logging import getLogger
+import pytz
+from pytz import timezone
 
 import dateparser
+import numpy as np
 import pandas as pd
-import pytz
-from pytz import timezone
+
 from pandas._libs.tslibs.timestamps import Timestamp
 
 from wetterdienst.core.core import Core
+from wetterdienst.dwd.metadata.column_names import DWDMetaColumns
+from wetterdienst.dwd.util import parse_datetime
 from wetterdienst.exceptions import NoParametersFound, StartDateEndDateError
 from wetterdienst.metadata.columns import Columns
+from wetterdienst.metadata.period import Period, PeriodType
+from wetterdienst.metadata.resolution import ResolutionType, Resolution, Frequency
 from wetterdienst.metadata.result import Result
 from wetterdienst.metadata.timezone import Timezone
-from wetterdienst.util.enumeration import parse_enumeration_from_template
-
+from wetterdienst.util.enumeration import (
+    parse_enumeration_from_template,
+)
+from wetterdienst.util.geo import Coordinates, derive_nearest_neighbours
 
 log = getLogger(__name__)
 
 
+EARTH_RADIUS_KM = 6371
+
+# TODO: move more attributes to __init__
+
+
 class PointDataCore(Core):
+    """Core for resolution based classes, not part of PointDataCore as it may not be
+    needed for metadata"""
+
+    @property
+    def resolution(self) -> Optional[Resolution]:
+        """ Resolution accessor"""
+        return self._resolution
+
+    @resolution.setter
+    def resolution(self, res) -> None:
+        # TODO: add functionality to parse arbitrary resolutions for cases where
+        #  resolution has to be determined based on returned data
+        if self._resolution_type in (ResolutionType.FIXED, ResolutionType.UNDEFINED):
+            self._resolution = res
+        else:
+            self._resolution = parse_enumeration_from_template(
+                res, self._resolution_base, Resolution
+            )
+
+    @property
+    @abstractmethod
+    def _resolution_base(self) -> Optional[Resolution]:
+        """ Optional enumeration for multiple resolutions """
+        pass
```

Comment:
Does the new architecture require derived classes to implement this? If so, we might want to raise an exception here as well as on other `@abstractmethod`s like
```python
raise NotImplementedError("Please implement this method")
```

Reply:
@abstractmethod automatically raises a NotImplementedError so I guess this is not necessary